pykern/pkasyncio.py

```diff
@@ -174,25 +169,31 @@ def _start(self):
         """
         while True:
             with self.__lock:
-                if self.destroyed:
+                if self.__destroyed:
                     return
             try:
                 m, a = self.__actions.get(**self.__get_args)
                 self.__actions.task_done()
             except queue.Empty:
                 m, a = self._on_loop_timeout(), None
             with self.__lock:
-                if self.destroyed:
-                    return
+                if self.__destroyed:
+                    break
                 # Do not need to check m, because only invalid when destroyed is True
-                if (m := self._dispatch_action(m, a)) is self._LOOP_END:
-                    return
+            if (m := self._dispatch_action(m, a)) is self._LOOP_END:
+                break
                 # Will be true if destroy called inside action (m)
-                if self.destroyed:
-                    return
+            with self.__lock:
+                if self.__destroyed:
+                    break
             # Action returned an external callback, which must occur outside lock
             if m:
                 self._dispatch_callback(m)
+            with self.__lock:
```

Comment:
I think this line should be outside the while loop. And, there is one case when the object is not destroyed but it exists, that's LOOP_END, which an action can return. It is used inside the upstream check, btw. If it is not destroyed, it should destroy itself, always.